src/systemd/meson.build

```diff
@@ -73,13 +77,17 @@ endif
 foreach header : _systemd_headers + _not_installed_headers + [libudev_h_path]
         foreach opt : opts
                 std_name = opt.length() == 2 ? '_'.join(opt[1].split(':')) : ''
-                name = ''.join(['cc-', header.split('/')[-1], '_', opt[0], std_name])
-                if want_tests != 'false'
-                        test(name,
-                             check_compilation_sh,
-                             args : cc.cmd_array() + ['-c', '-x'] + opt +
-                                    ['-Werror', '-include',
-                                     meson.current_source_dir() / header])
-                endif
+                test(''.join(['cc-', header.split('/')[-1], '_', opt[0], std_name]),
+                     env,
+                     suite : 'headers',
```

Comment:
That's nice. Thanks! I'd add that apart from running just the "headers" suite it makes it possible to exclude those tests. It should make it easier to run the whole systemd testsuite under Valgrind because some parts including the "headers" suite have to be skipped with `--no-suite check-compilation headers`. With `bash` it failed under Valgrind. Now that `bash` is gone it just doesn't make sense to run the compiler under Valgrind.